slide_analysis/UI/image_viewer.py

```diff
@@ -3,108 +3,128 @@
 # from PyQt5.QtWidgets import (QAction, QApplication, QFileDialog, QLabel,
 #                              QMainWindow, QMenu, QMessageBox, QScrollArea, QSizePolicy)
 #
+from PyQt5 import QtCore
+
 from PyQt5.QtCore import QDir, Qt
 from PyQt5.QtGui import QPalette, QPixmap
-from PyQt5.QtWidgets import QMainWindow, QFileDialog, QMessageBox, QAction, QApplication, QMenu
+from PyQt5.QtWidgets import *
 
 from slide_analysis.UI.image_helper import ImageHelper
+from slide_analysis.UI.tile_view_widget import TilePreviewPopup
 from slide_analysis.UI.ui_mainWindow import Ui_MainWindow
+from slide_analysis.UI.constants import SIMILAR_TILE_SIZE
 
 
 class ImageViewer(QMainWindow, Ui_MainWindow):
     def __init__(self):
         super(ImageViewer, self).__init__()
+        self.image_helper = None
         self.setupUi(self)
+        self.imageVerticalLayout = QBoxLayout(QBoxLayout.Down)
+        self.topImagesScrollAreaWidgetContents.setLayout(self.imageVerticalLayout)
+        self.topImagesScrollArea.setWidgetResizable(True)
+        self.image_popup_widget = None
         self.show()
 
         self.scale_factor = 0.0
 
-        # self.imageLabel = QLabel()
         self.imageLabel.setBackgroundRole(QPalette.Base)
         # self.imageLabel.setSizePolicy(QSizePolicy.Ignored, QSizePolicy.Ignored)
         # self.imageLabel.setScaledContents(True)
 
-        # self.scrollArea = QScrollArea()
-        # self.scrollArea.setBackgroundRole(QPalette.Dark)
-        # self.scrollArea.setWidget(self.imageLabel)
-        # self.setCentralWidget(self.scrollArea)
-
         self.create_actions()
         self.create_menus()
 
-        self.setWindowTitle("Image Viewer")
-        # self.resize(500, 400)
+    def eventFilter(self, source, event):
+        if event.type() == QtCore.QEvent.MouseMove:
```

Comment:
will this module ever contain some kind of `switch` on event type?
If yes create a switch
If no unite all this `if` into a single one

Reply:
python doesn't have switch statement. if-elif is the only option, which is now unnecessary because no other events are needed. But they might become handy later